src/main/java/io/cdap/plugin/http/source/batch/HttpBatchSourceConfig.java

```diff
@@ -29,6 +49,60 @@ protected HttpBatchSourceConfig(String referenceName) {
   @Override
   public void validate(FailureCollector failureCollector) {
     super.validate(failureCollector);
+    validateCredentials(failureCollector);
+  }
+
+  public void validateCredentials(FailureCollector collector) {
+    try {
+      if (getAuthType() == AuthType.OAUTH2) {
```

Comment:
We do have a service account auth type also, that also we have to add.

Reply:
Validation for service account already exists

Reply:
with proxy is it there?